src/Inline/Element/AbstractInline.php

```diff
@@ -26,38 +26,11 @@
     public $data = [];
 
     /**
-     * @var AbstractInline|null
+     * @return bool
      */
-    protected $parent;
-
-    /**
-     * @return AbstractBlock|AbstractInline|null
-     */
-    public function getParent()
+    public function isContainer()
     {
-        return $this->parent;
-    }
-
-    /**
-     * @param AbstractBlock|AbstractInline $parent
-     *
-     * @return $this
-     */
-    public function setParent($parent)
-    {
-        if (!$parent instanceof AbstractBlock && !$parent instanceof self) {
-            throw new \InvalidArgumentException(sprintf(
-                'Argument 1 passed to %s() must be an instance of %s or %s, instance of %s given',
-                __METHOD__,
-                'League\CommonMark\Block\Element\AbstractBlock',
-                'League\CommonMark\Block\Element\AbstractInline',
-                is_object($parent) ? get_class($parent) : gettype($parent)
-            ));
-        }
-
-        $this->parent = $parent;
-
-        return $this;
+        return false;
```

Comment:
Well, i tried to implement `ContainerNode` and failed... there is no way implement it without multiple inheritance... May be you have some thoughts about this problem?

Currently the problem is that every inline object already inherited from `Node`. Possible solution is to drop `AbstractInline` and `AbstractBlock` classes and make everything just a node. What do you think?